src/index.js

```diff
@@ -74,40 +66,150 @@ export default class CGUs extends events.EventEmitter {
+        return;
+      }
 
-      console.log(`${logPrefix} Fetched web page to ${snapshotPath}.`);
+      const snapshotId = await this.recordSnapshot({
+        snapshotContent: pageContent,
+        content: pageContent,
+        serviceId,
+        type,
+        logPrefix
+      });
 
       if (!snapshotId) {
-        return console.log(`${logPrefix} No changes, did not record snapshot.`);
+        return;
       }
 
-      console.log(`${logPrefix} Recorded snapshot with id ${snapshotId}.`);
+      return this.recordVersion({
+        snapshotContent: pageContent,
+        snapshotId,
+        serviceId,
+        documentDeclaration,
+        logPrefix,
+      });
+    } catch (error) {
+      console.error(`${logPrefix} Error:`, error.message);
+      this.emit('documentUpdateError', serviceId, type, error);
+    }
+  }
+
+  async refilterAndRecord(serviceToTrack) {
+    console.log('\nRefiltering documents… (it could take a while)');
 
-      const document = await filter(pageContent, documentDeclaration, this._serviceDeclarations[serviceId].filters);
+    const services = serviceToTrack ? { [serviceToTrack]: this._serviceDeclarations[serviceToTrack] } : this._serviceDeclarations;
+
+    const refilterAndRecordDocumentPromises = [];
+
+    Object.keys(services).forEach(serviceId => {
+      const { documents, name: serviceName } = this._serviceDeclarations[serviceId];
+      Object.keys(documents).forEach(type => {
+        refilterAndRecordDocumentPromises.push(this.refilterAndRecordDocument({
+          serviceId,
+          serviceName,
+          document: {
+            type,
+            ...documents[type]
+          }
+        }));
+      });
+    });
+
+    return Promise.all(refilterAndRecordDocumentPromises);
+  }
+
+  async refilterAndRecordDocument({ serviceId, serviceName, document: documentDeclaration }) {
+    const { type } = documentDeclaration;
+    const logPrefix = `[${serviceName}-${type}]`;
+    const { id: snapshotId, path: snapshotPath } = await history.getLastSnapshot(serviceId, type);
+
+    if (!snapshotId) {
+      return;
+    }
 
-      const { id: versionId, path: documentPath, isFirstRecord } = await recordVersion(serviceId, type, document, snapshotId);
-      if (versionId) {
-        const message = isFirstRecord
-          ? `${logPrefix} First version recorded in ${documentPath} with id ${versionId}.`
-          : `${logPrefix} Recorded version in ${documentPath} with id ${versionId}.`;
-        console.log(message);
-        this.emit(isFirstRecord ? 'documentAdded' : 'versionRecorded', serviceId, type, versionId);
-      } else {
-        console.log(`${logPrefix} No changes after filtering, did not record version.`);
+    let snapshotContent;
+    try {
+      snapshotContent = await fs.readFile(snapshotPath, { encoding: 'utf8' });
+    } catch (error) {
+      if (error.code === 'ENOENT') {
+        return console.error(`The snapshot file ${snapshotPath} does not exists.`);
```

Comment:
If the _do_ takes the _s_, there is no more left for the verb 🙂 
```suggestion
        return console.error(`The snapshot file ${snapshotPath} does not exist.`);
```